### PR TITLE
ci: use CURL for cloudsmith.sh and enable RPM upload

### PR DESCRIPTION
**Pull request by 6007135**
Per discussion on IRC and Discord.

This change requires adding three repository variables:
`CLOUDSMITH_ORG`
`CLOUDSMITH_OWNER`
`CLOUDSMITH_REPO`

### CODE REVIEW

**Comment by 6007135** [comment]
I just realized this failed on buster armv7 with a CURL certificate error [~]. Interestingly the other buster runners did not have this issue. I'll look into if I need to download and pass the latest CA cert manually. I would have assumed so, but only having the issue on one buster runner confuses me.

[~] https://github.com/6007135/tvheadend/actions/runs/7691769756

**Comment by 6007135** [comment]
Fixed by adding a `update-ca-certificates --fresh` per Christian's suggestion.

**Comment by 6007135** [comment]
All CI runners succeed on my fork (different commit hash because it was pre rebase).

https://github.com/6007135/tvheadend/actions/runs/7811271207

**Comment by Flole998** [comment]
You need to do something to line 36 of your script: The variable `package` is not assigned there yet (and all the other variables are always empty. Maybe move it down to the loop where the packages are actually pushed and do a `continue` (assuming that exists in bash, not sure).

**Comment by 6007135** [comment]
Good point. I should have caught that. Fixed.

File: support/cloudsmith.sh [hunk 1]
```diff
@@ -49,47 +49,51 @@ case $OSPREFIX$TARGET in
...
 for package in "${FILEARRAY[@]}"; do
+    EXTENSION=${package##*.}
+    PKGBASENAME=$(basename $package)
+    # dryrun exit is performed as late as possible to catch any earlier potential issues
     if [ $DRYRUN = "1" ]; then
-        echo "DRYRUN MODE: Skip pushing tvheadend/tvheadend/$OS/$TARGET $package"
-    else
-        python3 /usr/local/bin/cloudsmith push deb "tvheadend/tvheadend/$OS/$TARGET" $package || python /usr/local/bin/cloudsmith push deb "tvheadend/tvheadend/$OS/$TARGET" $package || cloudsmith push deb "tvheadend/tvheadend/$OS/$TARGET" $package
+        echo "DRYRUN MODE: Skip pushing $OS $TARGET package $package"
+        exit 0
```
Comment (Flole998) [inline support/cloudsmith.sh:77]: exit 0 is not really the right thing here, continue would be better as there might be multiple packages.

Comment (6007135) [reply support/cloudsmith.sh:77]: Ah, of course. Eventually I will learn to proofread a little better after cutting and pasting.

**Pull request merged**